lightning/src/ln/channel.rs

```diff
@@ -7866,70 +8006,17 @@ impl<SP: Deref> OutboundV1Channel<SP> where SP::Target: SignerProvider {
-		let mut monitor_signer = signer_provider.derive_channel_signer(self.context.channel_value_satoshis, self.context.channel_keys_id);
-		monitor_signer.provide_channel_parameters(&self.context.channel_transaction_parameters);
-		let channel_monitor = ChannelMonitor::new(self.context.secp_ctx.clone(), monitor_signer,
-		                                          shutdown_script, self.context.get_holder_selected_contest_delay(),
-		                                          &self.context.destination_script, (funding_txo, funding_txo_script),
-		                                          &self.context.channel_transaction_parameters, self.context.is_outbound(),
-		                                          funding_redeemscript.clone(), self.context.channel_value_satoshis,
-		                                          obscure_factor,
-		                                          holder_commitment_tx, best_block, self.context.counterparty_node_id, self.context.channel_id());
-		channel_monitor.provide_initial_counterparty_commitment_tx(
-			counterparty_initial_bitcoin_tx.txid, Vec::new(),
+		let channel_monitor = match self.initial_commitment_signed(
+			self.context.channel_id(),
+			msg.signature, counterparty_initial_bitcoin_tx.txid,
 			self.context.cur_counterparty_commitment_transaction_number,
-			self.context.counterparty_cur_commitment_point.unwrap(),
-			counterparty_initial_commitment_tx.feerate_per_kw(),
-			counterparty_initial_commitment_tx.to_broadcaster_value_sat(),
-			counterparty_initial_commitment_tx.to_countersignatory_value_sat(), logger);
-
-		assert!(!self.context.channel_state.is_monitor_update_in_progress()); // We have no had any monitor(s) yet to fail update!
-		if self.context.is_batch_funding() {
-			self.context.channel_state = ChannelState::AwaitingChannelReady(AwaitingChannelReadyFlags::WAITING_FOR_BATCH);
-		} else {
-			self.context.channel_state = ChannelState::AwaitingChannelReady(AwaitingChannelReadyFlags::new());
-		}
-		if self.context.holder_commitment_point.advance(&self.context.holder_signer, &self.context.secp_ctx, logger).is_err() {
-			// We only fail to advance our commitment point/number if we're currently
-			// waiting for our signer to unblock and provide a commitment point.
-			// We cannot send open_channel before this has occurred, so if we
-			// err here by the time we receive funding_signed, something has gone wrong.
-			debug_assert!(false, "We should be ready to advance our commitment point by the time we receive funding_signed");
-			return Err((self, ChannelError::close("Failed to advance holder commitment point".to_owned())));
-		}
-		self.context.cur_counterparty_commitment_transaction_number -= 1;
+			best_block, signer_provider, logger
+		) {
+			Ok(channel_monitor) => channel_monitor,
+			Err(err) => return Err((self, err)),
+		};
 
+		self.context_mut().cur_counterparty_commitment_transaction_number -= 1;
```

Comment:
No need to go through `context_mut()` here, `context`.

Reply:
Thanks. Missed this one when refactoring.